otsserver/backup.py

```diff
@@ -139,10 +140,10 @@ def write_disk_cache(self, chunk, bytes):
         cache_path = self.cache_path + '/' + chunk_path
         os.makedirs(cache_path, exist_ok=True)
         cache_file = cache_path + '/' + chunk_str
-        cache_file_tmp = cache_file + '.tmp'
-        with open(cache_file_tmp, 'wb') as fd:
-            fd.write(bytes)
-        os.rename(cache_file_tmp, cache_file)  # rename is atomic
+        temp_file = NamedTemporaryFile(delete=False)
```

Comment:
This will fail if `/tmp` happens to be on a difference filesystem than the calendar; use the `dir` argument to make sure the file is created in the correct directory.